src/mixins/filing-template-mixin.ts

```diff
@@ -345,6 +345,186 @@ export default class FilingTemplateMixin extends Mixins(AmalgamationMixin, DateM
     }
   }
 
+  /**
+   * Builds an continuation in filing from store data. Used when saving a filing.
+   * @returns the filing body to save
+   */
+  buildContinuationInFiling (): ContinuationInFilingIF {
+    function fixNullAddressType (orgPeople: OrgPersonIF[]): OrgPersonIF[] {
+      return orgPeople.map(p => {
+        if (p.deliveryAddress?.addressType === null) delete p.deliveryAddress.addressType
+        if (p.mailingAddress?.addressType === null) delete p.mailingAddress.addressType
+        return p
+      })
+    }
+
+    // Build the main filing.
+    const filing: ContinuationInFilingIF = {
+      header: {
+        name: FilingTypes.CONTINUATION_IN,
+        certifiedBy: this.getCertifyState.certifiedBy,
+        date: this.getCurrentDate,
+        filingId: this.getFilingId,
+        folioNumber: this.getFolioNumber,
+        isFutureEffective: this.getEffectiveDateTime.isFutureEffective
+      },
+      business: {
+        legalType: this.getEntityType,
+        identifier: this.getTempId
+      },
+      continuationIn: {
+        foreignJurisdiction: '',
+        nameRequest: {
+          legalType: this.getEntityType
+        },
+        nameTranslations: this.getNameTranslations,
+        offices: this.getDefineCompanyStep.officeAddresses,
+        contactPoint: {
+          email: this.getBusinessContact.email,
+          phone: this.getBusinessContact.phone,
+          // don't save extension if it's empty
+          ...this.getBusinessContact.extension
+            ? { extension: +this.getBusinessContact.extension }
+            : {}
+        },
+        parties: fixNullAddressType(this.getAddPeopleAndRoleStep.orgPeople)
+      }
+    }
+
+    // Add share structure data.
+    filing.continuationIn.shareStructure = {
+      shareClasses: this.getCreateShareStructureStep.shareClasses
+    }
```

Comment:
As this isn't conditional, might as well move it to line 391-ish.

(I'll do the same for amalgamations.)